### ISSUE TITLE
Bug: Auto Destroy cannot be disabled for ammunition

### ISSUE THREAD

**Issue by captainbob79** [foundryvtt/pf2e#8954]
After upgrading from v5.1.2 -> v5.2.3 I am no longer able to disable the Auto Destroy toggle for ammunition.

Steps to reproduce:
1. Add any ranged, ammunition weapon to a PC character sheet (tested with Arquebus and Shortbow)
2. Add the relevant ammunition type for the weapon(s) from the Compendium (tested with Arquebus rounds and arrows)
3. Edit the ammunition directly from the character sheet and disable Auto Destroy
4. Make a strike with the weapon from the actions tab, the ammunition count still decrements even with auto destroy disabled.

**Comment by captainbob79** [comment]
I wondered if that was the intended behavior. However, it is also not working that way either. Whether I have the toggle enabled or disabled, whenever I fire the last piece of ammunition that stack remains in my inventory with 0 quantity. 

Also, the behavior whenever I was on v5.1.2, it acted in the way I initially described, where the quantity only decremented if Auto Destroy was enabled. I know that part at least because we don't track ammunition in my game and I handled that be disabling Auto Destroy and carrying only 1 piece/quantity of ammunition. I have rolled my version back to v5.1.2 and confirmed that it behaves that way again.
